dartagnan/src/main/java/com/dat3m/dartagnan/program/event/lang/linux/LKMMLockRead.java

```diff
@@ -10,12 +10,12 @@
 public class LKMMLockRead extends Load {
 
 	public LKMMLockRead(Register register, IExpr lock) {
-		super(register, lock, Tag.Linux.MO_ACQUIRE);
+		super(register, lock, Linux.MO_ACQUIRE);
 		addFilters(RMW, Linux.LOCK_READ);
 	}
 
 	@Override
 	public String toString() {
-		return String.format("spin_lock_R(*%s)", address);
+		return String.format("%s <- spin_lock_R(*%s)", resultRegister, address);
```

Comment:
I would keep the old string representation which matches how the instruction is used in the litmus format, see e.g. [here](https://github.com/hernanponcedeleon/Dat3M/blob/master/litmus/LKMM/manual/locked/W%2BRW%2BRR%2BWR%2Bonces%2Blocked.litmus)

Reply:
But it results in code like this:
```
spin_lock_R(...)
assume(DUMMY_REG == 0);
```
where the assume is working on a register that you never see in the code.
I found it very confusing to be honest until I checked the spinlock event.

Reply:
Ohhh this is the read part (generated during compilation) of a spinlock. Then  it is ok (I thought it was the complete spinlock event). I will merge once the CI passes